src/openzaak/components/zaken/tests/test_zaakinformatieobjecten.py

```diff
@@ -204,3 +213,68 @@ def test_delete(self):
         # Relation is gone, zaak still exists.
         self.assertFalse(ZaakInformatieObject.objects.exists())
         self.assertTrue(Zaak.objects.exists())
+
+
+@tag("external-urls")
+class ExternalDocumentsAPITests(JWTAuthMixin, APITestCase):
+    heeft_alle_autorisaties = True
+
+    # TODO: validation tests with bad inputs
+
+    def test_relate_external_document(self):
+        base = "https://external.documenten.nl/api/v1/"
+        document = f"{base}enkelvoudiginformatieobjecten/{uuid.uuid4()}"
+        zio_type = ZaakInformatieobjectTypeFactory.create(
+            informatieobjecttype__concept=False, zaaktype__concept=False
+        )
+        zaak = ZaakFactory.create(zaaktype=zio_type.zaaktype)
+        zaak_url = f"http://testserver{reverse(zaak)}"
+        eio_response = get_eio_response(
+            document,
+            informatieobjecttype=f"http://testserver{reverse(zio_type.informatieobjecttype)}",
+        )
+
+        with self.subTest(section="zio-create"):
+            with requests_mock.Mocker() as m:
+                m.get(document, json=eio_response)
+                m.post(
+                    "https://external.documenten.nl/api/v1/objectinformatieobjecten",
+                    json=get_oio_response(document, zaak_url),
+                    status_code=201,
+                )
+
+                response = self.client.post(
+                    reverse(ZaakInformatieObject),
+                    {"zaak": zaak_url, "informatieobject": document},
+                )
+
+            self.assertEqual(
+                response.status_code, status.HTTP_201_CREATED, response.data
+            )
```

Comment:
Also check if the OIO is not created locally.

Reply:
Added